lib/gems/pending/util/mount/miq_local_mount_session.rb

```diff
@@ -0,0 +1,37 @@
+require 'util/mount/miq_generic_mount_session'
+
+# MiqLocalMountSession is meant to be a representation of the local file system
+# that conforms to the same interface as MiqLocalMountSession (and by proxy,
+# MiqFileSystem::Interface).
+#
+# See MiqGenericMountSession for info on methods available.
+class MiqLocalMountSession < MiqGenericMountSession
+  def self.uri_scheme
+    "file".freeze
+  end
+
+  # no-op these since they are not relavent to the local file system
+  #
+  # rubocop:disable Style/SingleLineMethods, Layout/EmptyLineBetweenDefs
+  def connect; end      # :nodoc:
+  def disconnect; end   # :nodoc:
+  def mount_share; end  # :nodoc:
+  # rubocop:enable Style/SingleLineMethods, Layout/EmptyLineBetweenDefs
+
+  def relative_to_mount(dest_uri) # :nodoc:
+    dest_uri
+  end
+
+  # :nodoc:
+  #
+  # This is an override to allow methods that make use of `@mnt_point` to
+  # behave as we would expect them to.
+  #
+  # Determine the mount point relative to the destination file, and just remove
+  # the portions of the expanded path that already exist in the
+  # `remote_file_path` (might be everything if `remote_file_path` is a absolute
+  # path)
+  def mnt_point
+    @mnt_point ||= File.expand_path(remote_file_path).sub(remote_file_path, "")
```

Comment:
After some thinking and looking at how to implement this in `EvmDatabaseOps`, this is a bad way to implement this via `memoization`.

Part of the reason to set `@mnt_point` was to prevent changes to `MiqGenericMountSession`, which doesn't use the accessor, but `@mnt_point` directly in a lot of spots.  But, there are plenty of times where `remote_file_path` will not be set, and so generating `@mnt_point` in this fashion will cause issues (possibly just error out) if it is set and memoized before a `#add` call.

I have some ideas on how to do this better, but know that this implementation is probably going to change.